Composer/packages/client/src/components/ManageQNA/ManageQNA.tsx

```diff
@@ -3,706 +3,173 @@
-          <p>
-            <label css={summaryLabelStyles}>{formatMessage('Key')}</label>
-            {localRootQNAKey}
-          </p>
-          <p>
-            <label css={summaryLabelStyles}>{formatMessage('Region')}</label>
-            {localRootQNARegion}
-          </p>
-        </div>
-      );
-      setOutcomeError(false);
+          {
+            name: 'UserAppInsightsAppId',
+            value: userAppInsightsAppId,
+          },
+          {
+            name: 'PrimaryEndpointKey',
+            value: primaryEndpointKey,
+          },
+          {
+            name: 'SecondaryEndpointKey',
+            value: secondaryEndpointKey,
+          },
+          {
+            name: 'DefaultAnswer',
+            value: defaultAnswer,
+          },
+          {
+            name: 'QNAMAKER_EXTENSION_VERSION',
+            value: QNAMAKER_EXTENSION_VERSION,
+          },
+        ],
+      },
+      enabled: true,
+    });
+
+    // Create qna account
+    const cognitiveServicesManagementClient = new CognitiveServicesManagementClient(tokenCredentials, subscriptionId);
+    await cognitiveServicesManagementClient.accounts.create(resourceGroupName, qnaMakerServiceName, {
+      kind: 'QnAMaker',
+      sku: {
+        name: 'F0',
+      },
+      location: region,
+      properties: {
+        apiProperties: {
+          qnaRuntimeEndpoint: `https://${webAppResult.hostNames?.[0]}`,
+        },
+      },
+    });
 
-      setCurrentPage(3);
-    } else if (nextAction === 'handoff') {
-      setShowHandoff(true);
-      props.onDismiss();
+    const keys = await cognitiveServicesManagementClient.accounts.listKeys(resourceGroupName, qnaMakerServiceName);
+    if (!keys?.key1) {
+      throw new Error('No key found for newly created authoring resource');
     } else {
-      setCurrentPage(2);
+      return keys.key1;
     }
   };
 
-  const iconStyles = { marginRight: '8px' };
-  const onRenderOption = (option) => {
-    return (
-      <div>
-        {option.data && option.data.icon && (
-          <Icon aria-hidden="true" iconName={option.data.icon} style={iconStyles} title={option.data.icon} />
-        )}
-        <span>{option.text}</span>
-      </div>
-    );
-  };
-
   return (
-    <Fragment>
-      {showAuthDialog && (
-        <AuthDialog
-          needGraph={false}
-          next={hasAuth}
-          onDismiss={() => {
-            setShowAuthDialog(false);
-          }}
-        />
+    <ManageService
+      createService={createService}
+      handoffInstructions={formatMessage(
+        'I am working on a Microsoft Bot Framework project, and I now require some Azure resources to be created. Please follow the instructions below to create these resources and provide them to me.\n\n1. Using the Azure portal, please create a QnAMaker resource on my behalf.\n2. Once provisioned, securely share the resulting credentials with me as described in the link below.\n\nDetailed instructions:\nhttps://aka.ms/bfcomposerhandoffqnamaker'
```

Comment:
Since this string is mostly the same in all three of these subpages, could we refactor it as well into the ManageService component and just provide the URL as a prop? It looks like you already have the service name coming as one.

Reply:
I thought about that, but I think there is a high degree of likelihood that these instructions DO become more differentiated as we move forward.  Would prefer to let the full text be specified for now.

Reply:
Can we at least refactor out the first half of it, up to the "provide them to me"? That seems unlikely to change.